catalog-rest-service/src/main/java/org/openmetadata/catalog/secrets/SecretsManagerMigrationService.java

```diff
@@ -0,0 +1,307 @@
+          .getData().stream()
+          .map(ServiceEntityInterface.class::cast)
+          .filter(
+              service ->
+                  !Objects.isNull(service.getConnection()) && !Objects.isNull(service.getConnection().getConfig()))
+          .collect(Collectors.toList());
+    } catch (IOException e) {
+      throw new SecretsManagerMigrationException(e.getMessage(), e.getCause());
+    }
+  }
+
+  private void migrateIngestionPipelines(IngestionPipeline ingestionPipeline) {
+    try {
+      IngestionPipeline ingestion = ingestionPipelineRepository.dao.findEntityById(ingestionPipeline.getId());
+      ingestion.getOpenMetadataServerConnection().setSecurityConfig(null);
+      ingestionPipelineRepository.dao.update(ingestion);
+    } catch (IOException e) {
+      throw new SecretsManagerMigrationException(e.getMessage(), e.getCause());
+    }
+  }
+
+  private List<IngestionPipeline> retrieveNotStoredIngestionPipelines() {
+    try {
+      return ingestionPipelineRepository
+          .listAfter(
+              null,
+              EntityUtil.Fields.EMPTY_FIELDS,
+              new ListFilter(),
+              ingestionPipelineRepository.dao.listCount(new ListFilter()),
+              null)
+          .getData().stream()
+          .filter(
+              ingestionPipeline ->
+                  !Objects.isNull(ingestionPipeline.getOpenMetadataServerConnection())
+                      && !Objects.isNull(ingestionPipeline.getOpenMetadataServerConnection().getSecurityConfig()))
+          .collect(Collectors.toList());
+    } catch (IOException e) {
+      throw new SecretsManagerMigrationException(e.getMessage(), e.getCause());
+    }
+  }
+
+  /** This method delete all the change events which could contain connection config parameters for services */
+  private void deleteChangeEventsForServices() {
+    connectionTypeRepositoriesMap.values().stream()
+        .map(ServiceEntityRepository::getServiceType)
+        .forEach(
+            serviceType -> {
+              try {
+                changeEventRepository.deleteAll(
+                    Entity.class
+                        .getField(serviceType.value().toUpperCase(Locale.ROOT) + "_SERVICE")
+                        .get(Entity.class)
+                        .toString());
+              } catch (NoSuchFieldException | IOException | IllegalAccessException e) {
+                throw new SecretsManagerMigrationException(e.getMessage(), e.getCause());
+              }
+            });
+  }
+
+  /**
+   * This method delete all the change events which could contain auth provider config parameters for ingestion
+   * pipelines
+   */
+  private void deleteChangeEventsForIngestionPipelines() {
+    ChangeEventRepository changeEventRepository = retrieveChangeEventRepository();
+    try {
+      changeEventRepository.deleteAll(Entity.INGESTION_PIPELINE);
+    } catch (IOException e) {
+      throw new SecretsManagerMigrationException(e.getMessage(), e.getCause());
```

Comment:
do we want to throw an exception or log silently and move on

Reply:
We should throw an exception instead of leaving the database in an inconsistent state and moving on. is it OK?